mu4e: Add possibility to forward messages as attachments

KrzysiekJ opened the pull request:
This commit adds a global variable `mu4e-compose-forward-as-attachment`. To enable choosing forwarding method on a per-message basis would probably require either:

* changing the mu server backend so that it distinguishes between inline forwarding and forwarding as attachment;
* changing the mu server backend so that it doesn’t return attachments at all and making both inline and as attachment forwarding via MIME (and also making mu4e actually display MIME-enclosed inline emails).

djcb commented on mu4e/mu4e-compose.el:
```diff
@@ -190,20 +190,25 @@ place to do that."
+  (mu4e-compose-attach-message mu4e-captured-message))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
```
Ah, the above is a nice re-factoring

djcb commented on mu4e/mu4e-compose.el:
```diff
@@ -514,7 +519,7 @@ automatically encrypt that reply."
 
-(defun* mu4e~compose-handler (compose-type &optional original-msg includes)
+(defun* mu4e~compose-handler (compose-type &optional original-msg includes as-attachment)
   "Create a new draft message, or open an existing one.
```
In this one (and the other), don't pass 'as-attachment' as a parameter; rather, use the value of mu4e-compose-forward-as-attachment directly in the function, so only `mu4e~compose-handler` and `mu4e~draft-forward-construct` need changes.

djcb commented:
Ah, thanks, looks interesting. See my comments in the code.

KrzysiekJ commented:
I’ve updated the commit to use global variables (previous version was deliberately not doing it to make the code more pure) and also removed adding two newlines to message body when forwarding as attachment.

djcb commented:
Merged, thanks!
